Dockerfile

```diff
@@ -23,78 +23,80 @@ RUN curl https://packages.microsoft.com/keys/microsoft.asc | apt-key add - \
 RUN wget https://download.java.net/java/GA/jdk13.0.2/d4173c853231432d94f001e99d882ca7/8/GPL/openjdk-13.0.2_linux-x64_bin.tar.gz \
-	&& mkdir $JAVA_HOME \
-	&& tar xv --file openjdk-13*_bin.tar.gz --directory "$JAVA_HOME" --no-same-owner --strip-components 1 \
-	&& find "$JAVA_HOME/lib" -name '*.so' -exec dirname '{}' ';' | sort -u > /etc/ld.so.conf.d/docker-openjdk.conf \
-	&& ldconfig
+    && mkdir $JAVA_HOME \
+    && tar xv --file openjdk-13*_bin.tar.gz --directory "$JAVA_HOME" --no-same-owner --strip-components 1 \
+    && find "$JAVA_HOME/lib" -name '*.so' -exec dirname '{}' ';' | sort -u > /etc/ld.so.conf.d/docker-openjdk.conf \
+    && ldconfig
 
 # Lets always use latest pip?
 RUN /usr/local/bin/python -m pip install --upgrade pip
 
-# Install some commonly used packages and the Python application
-RUN pip3 install --use-feature=2020-resolver --no-cache-dir --upgrade --force-reinstall \
-		avro \
-		fastavro \
-		beautifulsoup4 \
-		bokeh \
-		cloudpickle \
-		colorama \
-		cython \
-		dill \
+RUN mkdir /home/default \	
+	&& python3 -m venv $VIRTUAL_ENV \	
+    && python3 -m pip install --upgrade pip \
+    && pip3 install --use-feature=2020-resolver --no-cache-dir --upgrade --force-reinstall \
+        avro \
+        fastavro \
+        beautifulsoup4 \
+        bokeh \
+        cloudpickle \
+        colorama \
+        cython \
+        dill \
         httplib2 \
         ipython \
         flake8 \
         future \
         h5py \
-        idna\<2.10\
         matplotlib \
         mlflow \
         numba \
         numexpr \
         numpy \
         nose \
-        pandas\<1.0\
+        pandas\<1.2 \
         patsy \
         pymongo \
         pyodbc \
         PyYaml \
         pytest \
         qgrid \
-        requests\<2.24.0 \
         seaborn \
         snowflake-connector-python[pandas] \
         scipy \
         scikit-learn \
         scikit-image \
-        sqlalchemy\<=1.3.13\
+        sqlalchemy\
         statsmodels \
         sympy \
         tabulate \
         tqdm \
         vincent \
-        xlrd
+        xlrd \
+	&& chown :users -R /home/default \
+	&& chmod a+rwx -R /home/default 
 
 RUN pip3 install --no-cache-dir --upgrade --force-reinstall git+git://github.com/keboola/python-docker-application.git@2.2.0 \
     && pip3 install --no-cache-dir --find-links https://h2o-release.s3.amazonaws.com/h2o/latest_stable_Py.html h2o \
-	&& pip3 install --use-feature=2020-resolver --no-cache-dir --upgrade --force-reinstall \
+    && pip3 install --use-feature=2020-resolver --no-cache-dir --upgrade --force-reinstall \
         git+git://github.com/keboola/sapi-python-client.git@0.4.0 \
-        requests\<2.24.0 \
-        azure-storage-blob\<12.0.0 \ 
-        cffi\<1.14 \
-        cryptography\<3.0.0 \
-        boto3\<1.12 \
-        idna\<2.10 \
-        pytz\<2021.0 \
         keboola.component \
-    && mkdir -p /root/.cache/snowflake/
+        chardet\<4 \
+        certifi\<2021.0.0 \

```

Comment:
sorry, what is certifi?

Reply:
also, why get rid of boto3 and az blob storage?  I guess cuz they are included in the keboola.component?

Reply:
fixed